src/rez/cli/pip.py

```diff
@@ -40,6 +42,15 @@ def command(opts, parser, extra_arg_groups=None):
         p.wait()
         return
 
+    if opts.pip_ver:
+        with warnings.catch_warnings():
```

Comment:
I feel like we should just trigger the warning here, and leave it up to elsewhere to set the filter. Since this is CLI code, we could safely just set to `always` in cli._main or something. The reason I say is because there are a bunch of other places where deprecation warnings should go, that aren't necessarily in CLi code. In those cases you would not be wanting to set the warnings filter - that would be up to the host application to set.

We might also consider wrapping this with a utils.deprecation_warning() func.

Reply:
ps - I also think we should do this in a separate PR though, so happy to leave this as-is

Reply:
My thinking was that if we always let warnings be printed out, it could cause some noise. That's nice when doing development and when running tests, but it's not so nice for a user that just wants to use rez without getting warnings from other libs or python itself for example. That's why I haven't enabled them globally.

I also thought about a deprecation warning function but I wasn't sure if you would have like it since right now there is no deprecation policy in Rez (For example, see the policy of pip https://pip.pypa.io/en/stable/development/release-process/#deprecation-policy), as everything as to stay backward compatible... Introducing deprecation warnings would mean we would start to really deprecate things and break backward compatibility in some places... So I wasn't sure. But if we are fine with this (I am fully fine with breaking compatibility when necessary and when well documented with proper error messages/warnings that points to the right documentation), I'm all for a common function.